akka-http/src/main/scala/akka/http/scaladsl/server/directives/ReverseProxyDirectives.scala

```diff
@@ -140,3 +151,13 @@ case class CachedHostConnectionPoolRequestExecutor(implicit val system: ActorSys
   }
 }
 
+case class ReverseProxyConfig(
```

Comment:
Perhaps make it not a case class? They're nearly impossible to evolve in a binary compatible way.

See here for inspiration https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/javadsl/settings/ServerSettings.scala#L23

Hide it's fields behind an interface and create an `object ReverseProxyConfig` that serves as factory.
Impl can be a case class, but we need to hide it from users.